mono/metadata/cominterop.c

```diff
@@ -540,6 +541,30 @@ cominterop_com_visible (MonoClass* klass)
 
 }
 
+static guint32
+cominterop_class_iface_type (MonoClass* klass)
+{
+	ERROR_DECL (error);
+	static MonoClass *ClassInterfaceAttribute = NULL;
+	MonoCustomAttrInfo *cinfo;
+	guint32 result = 2;
+
+	if (!ClassInterfaceAttribute)
```

Comment:
Does this initialization of static need to be thread safe? maybe better to do it in mono_cominterop_init.

Reply:
I don't see any documented threading or locking requirement in mono_class_load_from_name. Maybe there is one that I'm not aware of. I do see there's a GENERATE_GET_CLASS_WITH_CACHE macro that seems designed for this purpose, so it's probably better to use that.

Reply:
Since this is a static maybe it is better to but it together with other statics in this file and init it as part of mono_cominterop_init. That will  make sure it is setup during startup and not during runtime. If that's not possible you might need to do something similar as done int mono_class_try_get_com_object_class. However I see that similar pattern is used on some other places as well so this source is not consistent so maybe it should be adjusted to follow similar pattern as used in other sources getting access to corlib classes.